### ISSUE TITLE
Cannot run PSM(0)

### ISSUE THREAD

**Issue by samiles** [thiagoalessio/tesseract-ocr-for-php#58]
Hi, I'm trying to run Tesseract in _Orientation and script detection (OSD) only_ mode (PSM 0).

On the command line, setting PSM 0 works and produces something like:

```
Orientation: 0
Orientation in degrees: 0
Orientation confidence: 22.31
Script: 1
Script confidence: 36.67
```

I was hoping that the following code would produce that result, but instead it just gives a PSM 3 result of the OCR (default):

    $tess = (new TesseractOCR(storage_path('app/doc.jpg')));
    $tess->psm(0);
    $text = $tess->run();
    //Returns PSM 3 default OCR result

The other PSM modes (1-10) produce expected results, it's just PSM 0 which I can't get to work.

Is there a way to run PSM 0 and get the actual orientation result instead of the OCR? Is this an error, or is getting the OCR back for PSM 0 the expected result?

Thanks

**Comment by samiles** [comment]
Some more info:

    $tess->psm(0); //returns PSM 3 result
    $tess->psm('0'); //returns PSM 3 result
    $tess->psm(); //FAILS
    $tess->psm(''); //returns PSM 3 result
    $tess->psm('hello'); //returns CORRECT PSM 0 result!

So, for now as a 'workaround' I am passing a string to PSM and running:

    $tess = (new TesseractOCR(storage_path('app/doc.jpg')));
    $tess->psm('hello');
    $text = $tess->run();

    //Returns:
    Orientation: 0
    Orientation in degrees: 0
    Orientation confidence: 22.31
    Script: 1
    Script confidence: 36.67

This gives the correct answer so will do for now. Any ideas why `psm(0)` doesn't work?

**Comment by thiagoalessio** [comment]
hehe nasty little bug, thanks a lot for finding and reporting @samiles !
It is fixed now (c343176). You can install `dev-master` until I release a new version that contains the bugfix.
Thanks again ;D

**Comment by nickbe** [comment]
This command is still needed. Is it possible to fix this?
